docs/source/nlp/nemo_megatron/retro/retro_model.rst

```diff
@@ -1,469 +1,170 @@
 
-    python scripts/nlp_language_modeling/build_knn_map_index.py \
-        --input_file=/result/pubmed_eval_text_document  \
-        --tokenizer-library=megatron \
-        --tokenizer-type=GPT2BPETokenizer \
-        --merge-file=/dataset/gpt2-merges.txt \
-        --vocab-file=/dataset/gpt2-vocab.json \
-        --process_chunk_size=10000 \
-        --sentence_transformer_model=all-mpnet-base-v2 \
-        --batch_size=1024 \
-        --K_neighbors=50 \
-        --workers=2 \
-        --devices=0,1,2,3,4,5,6,7 \
-        --remove_duplicate \
-        --dedup_margin=70 \
-        --nprobe=100 \
-        --shard_id=0 \
-        --total_shards=10 \
-        --stage=1 \
-        --output_file=/dataset/pubmed_knn_shard0.save \
-        --faiss_index=/result/pubmed_faiss_final.index
-
-In this example, the training data is broken into ``total_shards`` shards, and the KNN index is calculated for the shard specified by ``shard_id``. 
-The result is saved to a file specified by ``output_file``. In the example above, 10 KNN sharding indexes are created.
-
-Use the ``remove_duplicate`` flag if the training data and retrieval data are the same to remove neighbors from the same document.
-
-Step 4.2: Merge KNN sharding index into final KNN index
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-An example script is:
+The main output of this stage are:
 
-.. code-block:: bash
+* ``train_<UNIQUE_HASH>``: directory containing retrieved neighbors for all training samples.
+* ``valid_<UNIQUE_HASH>``: directory containing retrieved neighbors for all validating samples.
 
-    python scripts/nlp_language_modeling/build_knn_map_index.py  \
-    --stage=2 \
-    --output_file=pubmed_knn_final.save \
-    --shard_index_input=pubmed_knn_shard
 
 
-Train NeMo RETRO Model
+Train RETRO Model
 -----------------------
 
-Once the training data, retrieval data, KNN index, and Faiss index are prepared, we are ready to train the RETRO model. In the NeMo implementation, 
-the RETRO model can be pre-trained with or without the `mu-Transfer <https://openreview.net/pdf?id=Bx6qKuBM2AD>`_ :cite:`nlp-retro-yang2022tensor` feature. We will introduce both ways.
-
+Once the data (include training samples and pre-retrieved neighbors) are prepared, we are ready to train the RETRO model. The training process will use the output directory from the data preparation step. We set the path to this directory at the ``retro.retro_project_dir`` argument. Many of the data hyperparameters will be retrieved from the ``config.json`` file in this directory, including data splits, sequence length, chunk length, number of training and validating samples, tokenizer, etc.
```

Comment:
Once the training samples, pre-retrieved neighbors, and other data are prepared, you are ready to train the RETRO model. The training process will use the output directory from the data preparation step.